src/constants-nls.ts

```diff
@@ -0,0 +1,24 @@
+import { localize } from "./localize";
+
+export const viewOutput: string = localize('viewOutput', 'View Output');
+export const previewDescription: string = localize('preview', '(Preview)');
+export const pythonNewModelPreview: string = localize('pythonNewModelPreview', 'Python (Programming Model V2)');
+export const emptyWorkspace: string = localize('emptyWorkspace', 'Your workspace folder looks empty, please navigate to the root directory of your project.');
+export const useEmulator: string = localize('useEmulator', 'Use Local Emulator');
+export const skipForNow: string = localize('skipForNow', 'Skip For Now');
+export const overwriteRemoteConnection = (a: string) => localize('overwriteRemoteConnection', 'We detected a different local connection setting for "{0}" than what was previously used. Would you like to overwrite your remote setting?', a);
+
+export const invalidAlphanumericWithHyphens: string = localize('invalidAlphanumericOrHyphen', `A name must consist of alphanumeric characters or '-', and must start and end with an alphanumeric character.`);
+export const invalidLowerCaseAlphanumericWithHyphens: string = localize('invalidLowerAlphanumericOrHyphen', `A name must consist of lower-case alphanumeric characters or '-', and must start and end with a lower-case alphanumeric character.`);
+
+export const invalidLength = (lowerLimitIncl?: string, upperLimitIncl?: string) => {
```

Comment:
The name is pretty non-descript when you see it invoked in the code. Could we go with `getInvalidLengthError` or something along those lines? 

Also, I'd prefer if numbers were the parameters (which get casted into strings). It feels more intuitive that way since `isValidLength` accepts numbers as parameters.